sesman/libscp/libscp_v0.c

```diff
@@ -351,6 +350,19 @@ scp_v0s_accept(struct SCP_CONNECTION *c, struct SCP_SESSION **s, int skipVchk)
             /* until syslog merge log_message(s_log, LOG_LEVEL_WARNING, "[v0:%d] connection aborted: error setting password", __LINE__); */
             return SCP_SERVER_STATE_INTERNAL_ERR;
         }
+        if (code == SCP_GW_CHAUTHTOK)
+        {
+            /* reading new password */
+            in_uint16_be(c->in_s, sz);
+            buf[sz] = '\0';
```

Comment:
I know it's an "internal communication", but it looks scary. What if xrdp and xrdp-sesman run on different hosts, and one of the hosts is compromised? I think the value should be validated.